src/smolagents/models.py

```diff
@@ -1161,6 +1161,183 @@ def create_client(self):
+        # Many Bedrock models do not allow conversations to start with the `assistant` role, so the default is set to `user/user`.
+        # This parameter is retained for future model implementations and extended support.
+        self.custom_role_conversions = custom_role_conversions or {
+            MessageRole.SYSTEM: MessageRole.USER,
+            MessageRole.ASSISTANT: MessageRole.USER,
+            MessageRole.TOOL_CALL: MessageRole.USER,
+            MessageRole.TOOL_RESPONSE: MessageRole.USER,
+        }
+
+        super().__init__(
+            model_id=model_id,
+            custom_role_conversions=self.custom_role_conversions,
+            flatten_messages_as_text=flatten_messages_as_text,
+            **self.custom_api_kwargs,
+        )
+
+    def _prepare_completion_kwargs(
+        self,
+        messages: List[Dict[str, str]],
+        stop_sequences: Optional[List[str]] = None,
+        grammar: Optional[str] = None,
+        tools_to_call_from: Optional[List[Tool]] = None,
+        custom_role_conversions: dict[str, str] | None = None,
+        convert_images_to_image_urls: bool = False,
+        **kwargs,
+    ) -> Dict:
+        """
+        Overrides the base method to handle Bedrock-specific configurations.
+
+        This implementation adapts the completion keyword arguments to align with
+        Bedrock's requirements, ensuring compatibility with its unique setup and
+        constraints.
+        """
+        messages = super()._prepare_completion_kwargs(
+            messages=messages,
+            stop_sequences=stop_sequences,
+            grammar=grammar,
+            tools_to_call_from=tools_to_call_from,
+            custom_role_conversions=custom_role_conversions,
+            convert_images_to_image_urls=convert_images_to_image_urls,
+            **kwargs,
+        )
+
+        if messages.get("toolConfig"):
+            # Not all models in Bedrock support `toolConfig`. Also, smolagents already include the tool call in the prompt,
+            # so adding `toolConfig` could cause conflicts. We remove it to avoid issues.
+            messages.pop("toolConfig")
+
+        # The Bedrock API does not support the `type` key in requests.
+        # This block of code modifies the object to meet Bedrock's requirements.
+        for message in messages.get("messages", []):
+            for content in message.get("content", []):
+                if "type" in content:
+                    del content["type"]
+
+        return {
+            "modelId": self.model_id,
+            **messages,
+        }
+
+    def create_client(self):
+        # A custom client might already exist
+        # If so, we use the previously created client to avoid overwriting custom configurations or instances.
+        if getattr(self, "client", None):
+            return self.client
+
+        try:
+            import boto3
+
+            return boto3.client("bedrock-runtime", self.client_kwargs)
```

Comment:
I think this should be instead:
```suggestion
            return boto3.client("bedrock-runtime", **self.client_kwargs)
```

Also, maybe better moving the return out of the try-except scope.

Reply:
Yep! Done.